### DISCUSSION TITLE
Cannot find module './lib/fs' on node 6

### DISCUSSION THREAD

**Discussion by SystemParadox** [parcel-bundler/parcel#2533] [General] [answered]
# 🐛 bug report

Running the 'hello world' example on the website with parcel 1.11.0, I get the following error:

```
$ ./node_modules/.bin/parcel index.html
(node:2605) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module './lib/fs'
```

Digging into this a bit more, the issue appears to be in `@parcel/fs/index.js`, which tries to use `./src/fs` on node 8 or `./lib/fs` for older versions. However, the installation only appears to have a `src` directory. There is no `lib` directory in `@parcel/fs`.

This appears to work correctly with parcel 1.10.0.

## 💁 Possible Solution

Has someone forgotten to include the lib directory in the npm package?

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |  1.11.0
| Node             | 6.16.0
| npm/Yarn         | 6.5.0
| Operating System | Debian Linux

**Comment by DeMoorJasper** [comment]
Are you using any parcel plugins?

Some Parcel plugins are using internal Parcel files and the Parcel update kinda broke those as fs and some other things got split into their own package now

**Comment by SystemParadox** [comment]
No I literally just added the 'hello' world example and did `npm install --save-dev parcel`.

**Comment by DeMoorJasper** [comment]
Try `npm install parcel-bundler --save-dev`, parcel is a clone of `parcel-bundler`.

**Comment by SystemParadox** [comment]
Same issue. The problem is with the `@parcel/fs` package, which is missing the `lib` directory.

**Comment by DeMoorJasper** [comment]
Ah damn than it’s a publishing issue.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Abhay07** [comment]
Upgrade your npm version to 8 or above.
then again install parcel-bundler

This fixed my issue.

**Comment by DeMoorJasper** [comment, marked as answer]
Parcel 2 no longer supports Node 6.
We recommend updating to **Node 10** or newer regardless of which Parcel version you're running to have the best experience.

**Comment by DuckJao** [comment]
This fixed my issue, too. Can we add this solved result to the "getting started" page? I'm one of who just follow the "hello world" example ,get the error, and find here! Put this solved result in the page and notice next user like "if got this error, update your node to higher than v8!" maybe a good idea.

**Comment by SystemParadox** [comment]
Or we could just make sure it gets published with the compiled versions and then it will work for everyone...

**Comment by kjeske** [comment]
Any chances to have it fixed soon?

**Comment by DeMoorJasper** [comment]
@kjeske Parcel 2 will drop Node 6 support and Parcel 1 has been feature locked for now and only bug fixes get published so I doubt this will get fixed.

**Comment by andrewdyke** [comment]
I'm using Parcel and everything is fine when running locally, but I have my site running on Netlify, and when a build is triggered on Netlify it runs on Node 6 and then I get "(node:1198) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module './lib/fs'" ... any ideas to how I can get Nellify to use a new version of Node?

**Comment by JohannesKlauss** [comment]
I am using Node 10.16.0 and issue still pops up for me.

**Comment by DHBuild002** [comment]
Hi Guys, 

I would like to add my two pennies in here.

I reinstalled Node and that upgraded it from 6+ to 10+. That appears to have resolved the issue I was experiencing. 

Installing Node again should help you out if you have an older version of it at the moment: https://nodejs.org/en/download/

Check your version using this command: node -v
